### ISSUE TITLE
Indentation: no support for module declarations

### ISSUE THREAD

**Issue by vivek-0509** [checkstyle/checkstyle#21160]
Parent Issue #21113

I have read check documentation: https://checkstyle.org/checks/misc/indentation.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

Note: module-info.java is only parseable after #8240, so this is reproduced on that branch.

```bash
vivek@Viveks-MacBook-Air checkstyle % cat > config.xml <<'EOF'
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
    "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
    "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="Indentation"/>
  </module>
</module>
EOF

vivek@Viveks-MacBook-Air checkstyle % cat > ind/module-info.java <<'EOF'
module com.example.app {
        requires java.base;
  exports com.example.api;
}
EOF

vivek@Viveks-MacBook-Air checkstyle % RUN_LOCALE="-Duser.language=en -Duser.country=US"
java $RUN_LOCALE -jar target/checkstyle-13.10.0-SNAPSHOT-all.jar -c config.xml ind/module-info.java

Starting audit...
Audit done.
```

The same kind of indentation mistakes inside a class are reported:

```bash
vivek@Viveks-MacBook-Air checkstyle % cat > A.java <<'EOF'
package com.example;

class A {
        int a = 1;
  int b = 2;
}
EOF

vivek@Viveks-MacBook-Air checkstyle % java $RUN_LOCALE -jar target/checkstyle-13.10.0-SNAPSHOT-all.jar -c config.xml A.java

Starting audit...
[ERROR] A.java:4:9: 'member def type' has incorrect indentation level 8, expected level should be 4. [Indentation]
[ERROR] A.java:5:3: 'member def type' has incorrect indentation level 2, expected level should be 4. [Indentation]
Audit done.
Checkstyle ends with 2 errors.
```

**Describe what you expect in detail.**

Directives inside a module declaration should be indented one level from the `module` keyword, the same way members of a class body are, so both lines of the module input should be reported.

Module bodies get no indentation enforcement at all today. `HandlerFactory` registers no handler for `MODULE_DEF` or `DIRECTIVE_BLOCK`, so nothing inside the body is visited.

Expected on the module input:

```
2:9: 'module directive' has incorrect indentation level 8, expected level should be 4.
3:3: 'module directive' has incorrect indentation level 2, expected level should be 4.
```

**Comment by sahitya0xsingh** [comment]
I am on it.
